pkg/logql/syntax/serialize_test.go

```diff
@@ -0,0 +1,93 @@
+package syntax
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestJSONSerializationRoundTrip(t *testing.T) {
```

Comment:
Do we have a reasonable amount of coverage with these tests?

Reply:
These are 153 test cases
```
> go test -v -run TestJSONSerializationParseTestCases ./pkg/logql/syntax/... | rg PASS | sort | uniq | wc -l
152
```

they are *the* cases for the parser. That said there might be some edge cases. I was planning to take some actual queries, sanitize them and the add them. If it's ok I'll do this afterwards as I need the visitor for the sanitizing.

Reply:
Thanks. I actually did find two errors https://github.com/grafana/loki/pull/11230